### PR TITLE
Fix missing reference

### PR DESCRIPTION
**Pull request by 0101**
This should fix issues with namespaces disappearing with in-memory cross-project references, which happened quite often with FCS solution when using live buffers (#14033).

The underlying issue was in incremental builder where it's assumed that computing the last bound model would evaluate all the other ones too, which is not necessarily the case, and then accessing them by `ValueOption.Value`. 

For some reason the diagnostics for this didn't show. They are reported from here: https://github.com/dotnet/fsharp/blob/fdba9745df33c20f86a2fef120b4d37df72d5d12/src/Compiler/Driver/CompilerImports.fs#L2221-L2225

One suspected location which could produce an "unlinked" graph node is here: 
https://github.com/dotnet/fsharp/blob/fdba9745df33c20f86a2fef120b4d37df72d5d12/src/Compiler/Service/IncrementalBuild.fs#L1101

Even though we already have the result, the internal `cachedResult` will not be automatically populated and `tryPeekValue` will return `None`. Will probably fix that in a separate PR.

But this doesn't have to be (the only) reason for the issue, so better just avoid using `.Value` altogether. Probably should get rid of this one as well: https://github.com/dotnet/fsharp/blob/fdba9745df33c20f86a2fef120b4d37df72d5d12/src/Compiler/Service/IncrementalBuild.fs#L365

### CODE REVIEW

**Comment by 0101** [comment]
> Tests? 🤔

Wasn't able to figure out exact repro steps that could be put into a test... 

If existing tests still pass, we're replacing an unsafe call to `.Value` with a `match`, should be an improvement 😄

File: src/Compiler/Service/IncrementalBuild.fs [hunk 1]
```diff
@@ -1058,11 +1058,17 @@ module IncrementalBuilderStateHelpers =
...
-            let boundModels =
-                boundModels.ToImmutable()
-                |> ImmutableArray.map (fun x -> x.TryPeekValue().Value)
+            let! boundModels =
+                boundModels
+                |> Seq.map (fun x ->
+                    match x.TryPeekValue() with
+                    | ValueSome v -> node.Return v
+                    // *Evaluating the last bound model doesn't always guarantee that all the other bound models are evaluated.
+                    | _ -> node.ReturnFrom(x.GetOrComputeValue()))
```
Comment (majocha) [inline src/Compiler/Service/IncrementalBuild.fs:1070]: Don't the two branches do almost the same thing now? GetOrCompute have a happy path that returns cached value?

**Comment by majocha** [comment]
> Wasn't able to figure out exact repro steps that could be put into a test...

I've seen it throw exception exactly once when running VS in debug mode. I thought Option.Value a bit sus 🤨 but maybe a fluke because of changing repo branches or something. Not easy to hit.

File: src/Compiler/Service/IncrementalBuild.fs (hunk 1 above)
Comment (0101) [reply src/Compiler/Service/IncrementalBuild.fs:1070]: It does, but the problem is when there is no cached value.

**Review by abonie — APPROVED** [review]

**Pull request merged**